public/app/plugins/panel/geomap/layers/data/photosLayer.tsx

```diff
@@ -0,0 +1,202 @@
+  EventBus,
+  PluginState,
+  FieldType,
+  Field,
+} from '@grafana/data';
+import Map from 'ol/Map';
+import { FeatureLike } from 'ol/Feature';
+import { getLocationMatchers } from 'app/features/geo/utils/location';
+import VectorLayer from 'ol/layer/Vector';
+import { FrameVectorSource } from 'app/features/geo/utils/frameVectorSource';
+import { Stroke, Style } from 'ol/style';
+import Photo from 'ol-ext/style/Photo';
+import { findField } from 'app/features/dimensions';
+
+// Configuration options for Circle overlays
+export interface PhotoConfig {
+  kind: 'square' | 'circle' | 'anchored' | 'folio';
+  border?: number; // Sets border width around images
+  shadow?: boolean; // Renders drop shadow behind images
+  crop?: boolean; // Crops images to fill shape
+  src?: string; // Image source field
+  radius?: number; // Image radius
+}
+
+const defaultOptions: PhotoConfig = {
+  kind: 'square',
+  border: 2,
+  shadow: true,
+  crop: true,
+  radius: 20,
+};
+
+export const PHOTOS_LAYER_ID = 'photos';
+
+// Used by default when nothing is configured
+export const defaultPhotosConfig: MapLayerOptions<PhotoConfig> = {
+  type: PHOTOS_LAYER_ID,
+  name: '', // will get replaced
+  config: defaultOptions,
+  location: {
+    mode: FrameGeometrySourceMode.Auto,
+  },
+  tooltip: true,
+};
+
+// TODO, should be a question mark or missing or something? - let's embed a simple base64 image?
+const unknownImage = 'https://cdn.iconscout.com/icon/free/png-256/question-mark-1768084-1502257.png';
+
+/**
+ * Map layer configuration for circle overlay
+ */
+export const photosLayer: MapLayerRegistryItem<PhotoConfig> = {
+  id: PHOTOS_LAYER_ID,
+  name: 'Photos',
+  description: 'Render photos at each data point',
+  isBaseMap: false,
+  showLocation: true,
+  hideOpacity: true,
+  state: PluginState.alpha,
+
+  /**
+   * Function that configures transformation and returns a transformer
+   * @param map
+   * @param options
+   * @param theme
+   */
+  create: async (map: Map, options: MapLayerOptions<PhotoConfig>, eventBus: EventBus, theme: GrafanaTheme2) => {
+    // Assert default values
+    const config = {
+      ...defaultOptions,
+      ...options?.config,
+    };
+
+    const location = await getLocationMatchers(options.location);
+    const source = new FrameVectorSource(location);
+    const vectorLayer = new VectorLayer({
+      source,
+    });
+
+    let images: string[] = [];
+
+    vectorLayer.setStyle((feature: FeatureLike) => {
+      let src = unknownImage;
+      if (images.length > 0) {
+        const idx = feature.get('rowIndex') as number;
+        src = images[idx] ?? unknownImage;
+      }
+      return new Style({
+        image: new Photo({
+          src,
+          radius: config.radius,
+          crop: config.crop,
+          kind: config.kind,
+          shadow: config.shadow,
+          stroke: new Stroke({
+            width: config.border ?? 0,
+            color: '#000', // TODO set border color from theme?
```

Comment:
🤔 I think just adding another option makes sense.  Dark grey will look fine on both dark+light basemaps -- if folks pick something else, then they can pick a color that makes sense also